include/quicr/detail/uintvar.h

```diff
@@ -47,84 +49,66 @@ namespace quicr {
         }
     }
 
-    using UintV = std::vector<uint8_t>;
-
-    /**
-     * @brief Get the byte size from variable length-integer
-     *
-     * @param uint_v_msbbyte     MSB byte of the variable length integer
-     *
-     * @returns the size in bytes of the variable length integer
-     */
-    inline uint8_t SizeofUintV(const uint8_t uint_v_msbbyte)
+    class UintVar
     {
-        if ((uint_v_msbbyte & 0xC0) == 0xC0) {
-            return 8;
-        } else if ((uint_v_msbbyte & 0x80) == 0x80) {
-            return 4;
-        } else if ((uint_v_msbbyte & 0x40) == 0x40) {
-            return 2;
-        } else {
-            return 1;
+      public:
+        constexpr UintVar(uint64_t value)
+          : _value{ SwapBytes(value) }
+        {
+            constexpr uint64_t kLen1 = (static_cast<uint64_t>(-1) << (64 - 6) >> (64 - 6));
+            constexpr uint64_t kLen2 = (static_cast<uint64_t>(-1) << (64 - 14) >> (64 - 14));
+            constexpr uint64_t kLen4 = (static_cast<uint64_t>(-1) << (64 - 30) >> (64 - 30));
+
+            if (static_cast<uint8_t>(_value) & 0xC0u) { // Check if invalid
+                throw std::invalid_argument("Value greater than uintvar maximum");
+            }
+
+            if (value > kLen4) { // 62 bit encoding (8 bytes)
+                _value |= 0xC0ull;
+            } else if (value > kLen2) { // 30 bit encoding (4 bytes)
+                _value >>= 32;
+                _value |= 0x80ull;
+            } else if (value > kLen1) { // 14 bit encoding (2 bytes)
+                _value >>= 48;
+                _value |= 0x40ull;
+            } else {
+                _value >>= 56;
+            }
         }
-    }
 
-    /**
-     * @brief Convert uint64_t to Variable-Length Integer
-     *
-     * @details Encode unsigned 64bit value to shorten wrire format per RFC9000 Section 16 (Variable-Length Integer
-     * Encoding)
-     *
-     * @param value         64bit value to convert
-     *
-     * @returns vector of encoded bytes or empty vector if value is invalid
-     */
-    inline UintV ToUintV(uint64_t value)
-    {
-        constexpr uint64_t kLen1 = (static_cast<uint64_t>(-1) << (64 - 6) >> (64 - 6));
-        constexpr uint64_t kLen2 = (static_cast<uint64_t>(-1) << (64 - 14) >> (64 - 14));
-        constexpr uint64_t kLen4 = (static_cast<uint64_t>(-1) << (64 - 30) >> (64 - 30));
+        UintVar(Span<const uint8_t> bytes)
+          : _value{ 0 }
+        {
+            if (bytes.empty() || bytes.size() > sizeof(uint64_t)) {
+                throw std::invalid_argument("Invalid number of bytes for uintvar");
+            }
 
-        if (value & (0xC0ull << 56)) { // Check if invalid
-            return {};
+            std::memcpy(&_value, bytes.data(), bytes.size());
```

Comment:
Dangerous to not use `UintVar::Size(bytes.front())` for the size.  Length could be incorrect.  For example, the data could have more trailing bytes that might be for the next uintvar.